script.js

```diff
@@ -0,0 +1,217 @@
+
+}
+
+function isNumber(word) {
+    const numbers = /\d/g;
+    const matchingResult = word.match(numbers);
+    return matchingResult !== null;
+}
+
+function isValidPassword(password) {
+    return isLettersNumber(password) &&
+        isUppercaseLetter(password) &&
+        isNumber(password);
+}
+
+console.log(isValidPassword('Password123'));
+
+// 9. In the getPasswordStrength.js file.
+
+// 10. Create the getTotalOrderCost function that returns the total cost of the order and accepts the following arguments:
+// - the base cost of the items
+// - whether the customer is a VIP member - they get a 5% discount
+// - how many loyalty points the customer wants to use - each one of them reduces the cost by a cent (the final cost should not be lower than $0)
+// - is the package going to be shipped internationally - shipping internationally costs $10 (otherwise, it costs $5)
+// Make sure to apply the VIP discount before the loyalty points. The VIP discount does not apply to shipping.
+
+function getTheCostForVIPMember(cost) {
+    return cost * 0.05;
+}
+
+function getLoyaltyPointsDiscount(points) {
+    return  points / 100;
+}
+
+function getShippedInternationally(shippingType) {
+    if (shippingType === 'international') {
+        return 10;
+    }
+    return 5;
+}
+
+function getTotalOrderCost(baseCost, isVipMember, loyaltyPoints, shippingType) {
+    const shippingCost = getShippedInternationally(shippingType);
+    if (isVipMember) {
+        return baseCost - getTheCostForVIPMember(baseCost) - getLoyaltyPointsDiscount(loyaltyPoints) + shippingCost;
+    }
+    return baseCost - getLoyaltyPointsDiscount(loyaltyPoints) + shippingCost;
+}
+
+console.log(getTotalOrderCost(100, '', 5,  'international'));
+
+// 11. Create the getTicketPrice function that returns the ticket price and accepts the following arguments:
+// - the base price of the ticket
+// - how many days are there until the show - buying tickets more than a month in advance provides a 10% discount
+// - is the show on the weekend - weekend shows are $15 more expensive
+// The discount applied if the tickets are bought a month in advance should decrease the $15 charged if the show plays on the weekend.
+
+function getDiscountByNumberOfDays(daysNumber) {
+    if (daysNumber > 30) {
+        return 0.10;
+    }
+    return 0;
+}
+
+function getTicketPrice(basePrice, daysUntilTheShow, isWeekendShows) {
+    const tenPercentDiscountOnBasePrice = getDiscountByNumberOfDays(daysUntilTheShow) * basePrice;
+    const basePriceAfterTenPercentDiscount = basePrice - tenPercentDiscountOnBasePrice;
+    const tenPercentDiscountOnFifteenDollarsCharged = 15 * getDiscountByNumberOfDays(daysUntilTheShow);
+    const fifteenDollarsChargedAfterTenPercentDiscount = 15 - tenPercentDiscountOnFifteenDollarsCharged;
+    if (isWeekendShows) {
+        return basePriceAfterTenPercentDiscount + fifteenDollarsChargedAfterTenPercentDiscount;
+    }
+    return basePrice - tenPercentDiscountOnBasePrice;
```

Comment:
This code is super complicated.

We want to apply the discount to the base price increased by the weekend show fee. The formula is quite simple:

(base price + weekend show fee) * discount

I suggest you start this exercise completely from scratch and rethink it.

Reply:
This code is super complicated.

We want to apply the discount to the base price increased by the weekend show fee. The formula is quite simple:

(base price + weekend show fee) * discount

I suggest you start this exercise completely from scratch and rethink it.